node/node.go

```diff
@@ -1295,32 +1294,34 @@ type validatedBlock struct {
 	vb *ledgercore.ValidatedBlock
 }
 
-// Block satisfies the agreement.ValidatedBlock interface.
-func (vb validatedBlock) Block() bookkeeping.Block {
-	blk := vb.vb.Block()
-	return blk
+// unfinishedBlock satisfies agreement.UnfinishedBlock
+type unfinishedBlock struct {
+	blk *ledgercore.UnfinishedBlock
```

Comment:
This is the type I don't understand.  Now that we have a `ledgercore.UnfinishedBlock` and I think you even removed `Block()` so it can't be mistaken for a proposable or validated block, why not use that type directly in `node` package?

Reply:
It seems to come down to this.  A `ledgercore.ValidatedBlock` has this:
```
// FinishBlock completes the block and returns a proposable block.
func (ub UnfinishedBlock) FinishBlock(s committee.Seed, proposer basics.Address, eligible bool) bookkeeping.Block {
	return ub.blk.WithProposer(s, proposer, eligible)
}
```
but it doesn't really return a proposable block, it returns a block. So this `FinishBlock` doesn't implement `agreement.UnfinishedBlock`. So we have a `ledgercore.UnfinishedBlock` that is not an `agreement.UnfinishedBlock`.

Seems like the root problem is having `ledgercore` and `agreement` wanting to both do a lot with blocks, but being unable to share anything amongst each other. So `node` creates all sort of similarly named, but subtly different shims to convince them to just do the obvious things - modify some fields in a block.

Would this be fixed by moving the various `*Block*` interfaces into `bookkeeping`? `agreement` and `ledgecore` don't have to be declare their own interfaces, just to forced `node` to bend them to each others' will, do they?

Reply:
I tried moving the interfaces into `bookkeeping` in a PR on top of this one. https://github.com/cce/go-algorand/pull/18
I think it's simpler because it requires few types and less wrapping, but I suppose it's debatable. What do you think?